### PR TITLE
#164798204  create user profile 

### PR DESCRIPTION
**Pull request by felixkiryowa**
#### What does this PR do?

- Enables user profiles to be created, listed and updated.

#### Description of Task to be completed?

- user's profile is created on registration
- a user can view their profile and the profile of other users
- a user can update their profile but not the profile of other users
 
#### How should this be manually tested?

- Fetch this branch `git fetch origin ft-create-user-profile-164798204`
- Checkout to the branch `git checkout ft-create-user-profile-164798204`
- Run `pipenv install` command to install packages
- Run `python manage.py migrate`
- Run `python manage.py runserver`

##### Endpoints: 
- List Users Profiles: GET http://127.0.0.1:8000/api/profiles/
- View a user's profile: GET http://127.0.0.1:8000/api/profiles/username
- Update a user's profile: PUT http://127.0.0.1:8000/api/profiles/username 

#### What are the relevant pivotal tracker stories?
[#164798204](https://www.pivotaltracker.com/story/show/164798204)

#### Postman Collection

[Run in Postman](https://app.getpostman.com/run-collection/e09d4ff28bce66c4742b)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -10,16 +10,43 @@
...
-django-cors-headers
-django-extensions
-django-heroku
-djangorestframework
-gunicorn
-jwt
-pipenv-to-requirements
-psycopg2-binary
-python-decouple
+asn1crypto==0.24.0
```
Comment (ja-odur) [inline requirements.txt]: @felixkiryowa please use `pipenv run pipenv_to_requirements` to generate the `requirements.txt` file

File: Pipfile [hunk 2]
```diff
@@ -20,6 +20,7 @@ coverage = "*"
 gunicorn = "*"
 django-heroku = "*"
 pipenv-to-requirements = "*"
+django-rest-swagger = "*"
```
Comment (ja-odur) [inline Pipfile]: @felixkiryowa how come you have this requirement

File: authors/apps/profiles/views.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    lookup_field = 'username'
+
+    def get_object(self):
+        return get_object_or_404(
+            self.get_queryset(), user__username=self.kwargs.get('username')
+        )
+
+    def put(self, request, *args, **kwargs):
+        profile = self.get_object()
+        profile = Profile.objects.get(username=kwargs["username"])
```
Comment (ja-odur) [inline authors/apps/profiles/views.py]: @felixkiryowa why do you have repeated definition of profile i.e. on line `43 and 44`

**Review by dorothyas — DISMISSED** [review]

**Review by dorothyas — DISMISSED** [review]

**Review by dannylwe — DISMISSED** [review]

File: authors/apps/profiles/models.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+"""
+Module to define user profile model
+"""
+from django.conf import settings
```
Comment (ezrogha) [inline authors/apps/profiles/models.py]: Hey @felixkiryowa & @dorothyas this import is unused,  please remove it

File: authors/apps/profiles/tests/test_data.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "image": "https://www.django-rest-framework.org/api-guide/joan.png"
+    
+}
+
+update_user_profile_data_with_wrong_url = {
+    "firstname": "Joan",
+    "lastname": "Nantukunda",
+    "bio": "I love working out",
+    "image": "http"
+}
```
Comment (ezrogha) [inline authors/apps/profiles/tests/test_data.py]: please a new line at the end of this file

File: authors/apps/profiles/urls.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+from django.urls import path
+
+from .views import ProfileCreateAPIView, ProfileDetailAPIView
+
+app_name = 'profiles'
+urlpatterns = [
+    path('profiles/<str:username>', ProfileDetailAPIView.as_view()),
+    path('profiles/', ProfileCreateAPIView.as_view()),
+]
```
Comment (ezrogha) [inline authors/apps/profiles/urls.py]: Also add a new line to the end of this file

**Review by ezrogha — CHANGES_REQUESTED** [review]
Great work, kindly make those few changes

**Review by ezrogha — APPROVED** [review]

**Review by dorothyas — APPROVED** [review]

**Review by ja-odur — APPROVED** [review]
LGTM

**Pull request merged**